src/include/processor/operator/persistent/reader.h

```diff
@@ -40,13 +50,18 @@ class Reader : public PhysicalOperator {
     void getNextNodeGroupInParallel();
 
 private:
-    ReaderInfo readerInfo;
+    std::unique_ptr<ReaderInfo> info;
     std::shared_ptr<storage::ReaderSharedState> sharedState;
-    std::unique_ptr<common::DataChunk> dataChunkToRead;
+
     storage::LeftArrowArrays leftArrowArrays;
 
+    std::unique_ptr<common::DataChunk> dataChunk = nullptr;
+    common::ValueVector* nodeOffsetVector = nullptr;
```

Comment:
Do we need this field `nodeOffsetVector`? I would just fetch it dynamically from `resultSet`.

Reply:
This is to be consistent across our operator implementation